lib/gnat.ex

```diff
@@ -90,10 +91,29 @@ defmodule Gnat do
   """
   def unsub(pid, sid, opts \\ []), do: GenServer.call(pid, {:unsub, sid, opts})
 
+  @doc """
+  Ping the NATS server
+
+  This correlates to the [PING](http://nats.io/documentation/internals/nats-protocol/#PINGPONG) command in the NATS protocol.
+  If the NATS server responds with a PONG message this function will return `:ok`
+  ```
+  {:ok, gnat} = Gnat.start_link()
+  :ok = Gnat.ping(gnat)
+  ```
+  """
+  def ping(pid) do
```

Comment:
Exposing this as usable behavior seems odd to me. What happens for instance if two different clients both send call `Gnat.ping` close to the same time? The `state.pinger` would end up holding the address of the second client and the first one would timeout waiting for a `PONG`.

I think it probably makes sense for the `Gnat` actor to send a `PING` every so often and expect a response, but I don't understand the purpose of making it a callable behavior. Maybe I'm missing something?